src/main/java/dao/H2Factory/ConnectionProvider.java

```diff
@@ -0,0 +1,56 @@
+package dao.H2Factory;
+
+import dao.DaoException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class ConnectionProvider {
+
+    private Connection databaseConnection;
```

Comment:
I'm not sure that I understand it clearly. I will get troubles when many users will use my app?